lib/acme/resources/challenges/dvsni01.rb

```diff
@@ -0,0 +1,12 @@
+class Acme::Resources::Challenges::DVSNI01 < Acme::Resources::Challenges::Base
+
+  def sni_name
+    hd = crypto.digest.hexdigest(authorization_key)
+    "#{hd[0..31]}.#{hd[32..64]}.acme.invalid"
+  end
+
+  def request_verification
+    response = client.connection.post(@uri, { resource: 'challenge', type: 'dvsni-01', keyAuthorization: authorization_key })
```

Comment:
Did you test this? The old and removed name was `dvsni`: https://github.com/letsencrypt/boulder/pull/1247/files#diff-8be8b6353017e06063c146243e3eb2d2L71 but I can't find `dvsni-01` anywhere. The new name is `tls-sni-01`: https://github.com/letsencrypt/boulder/blob/6a575430f1cd4a27a498937a4fd12d258fe26a7c/core/objects.go#L71 and https://github.com/letsencrypt/boulder/blob/6a575430f1cd4a27a498937a4fd12d258fe26a7c/va/validation-authority.go#L585-L586